src/tests/test_magic.py

```diff
@@ -603,3 +603,31 @@ def test_columns_with_dollar_sign(ip_empty):
     html = result.result._repr_html_()
 
     assert "$2" in html
+
+
+def test_save_with(ip):
+    # First Query
+    ip.run_cell(
+        "%sql --save shakespeare SELECT * FROM author WHERE last_name = 'Shakespeare'"
```

Comment:
I think there are only 2 records in this table, so running `SELECT * FROM author WHERE last_name = 'Shakespeare'"` will always return 1 record, so the limit of 10 is not needed and we might miss some edge cases here (e.g: 2nd and 3rd queries return different results, None, etc...)

Maybe we can test it with more queries and verify we are getting the expected results for each. we can do it with `@pytest.mark.parametrize` and basically keep the same structure.

Also, I would add some error tests, like what happens when there is no such table, bad input, etc...

Reply:
I agree we should use more data to test it, fixed in [7d91bf8](https://github.com/ploomber/jupysql/pull/238/commits/7d91bf87fdeae784f9612d774480599b431d9b69)

For the error tests, let me know what you think we should cover more?

Reply:
I think let's test the most common use cases: bad table inputs (for --save and --with) and bad queries. If anything else comes up in your mind go for it.